blocks/blogheader/blogheader.js

```diff
@@ -15,6 +18,97 @@ function toggleMenu(nav, desktop) {
   nav.style.display = expand ? 'table' : 'none';
 }
 
+async function getLocaleBlogContents() {
+  if (window.serviceNowBlogContents) {
+    return window.serviceNowBlogContents;
+  }
+
+  const locale = getLocale();
+  window.serviceNowBlogContents = ffetch(`${BLOG_QUERY_INDEX}`)
+    .sheet('blogs-content')
+    .filter((entry) => entry.locale === locale)
+    .all();
+  return window.serviceNowBlogContents;
+}
+
+function markSearchTerms(link, searchTerms) {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const term of searchTerms) {
+    const regex = new RegExp(`(${term})`, 'gi');
+    link.innerHTML = link.innerHTML.replace(regex, '<mark>$1</mark>');
+  }
+}
+
+function indicateSearch(block) {
+  block.querySelector('form').classList.add('searching');
+}
+
+function unindicateSearch(block) {
+  block.querySelector('form').classList.remove('searching');
+}
+
+function focusSearch(block) {
+  block.querySelector('div.search-container').classList.add('search-active');
+}
+
+function blurSearch(block) {
+  block.querySelector('div.search-container').classList.remove('search-active');
+  const searchResults = block.querySelector('.search-results');
+  searchResults.style.display = 'none';
+  block.querySelector('input').value = '';
+  unindicateSearch(block);
+}
+
+async function handleSearch(block) {
+  const searchValue = block.querySelector('input').value;
+  const searchResults = block.querySelector('.search-results');
+
+  focusSearch(block);
+
+  if (searchValue.length < 3) {
+    searchResults.innerHTML = '';
+    searchResults.style.display = 'none';
+    return;
+  }
+
+  indicateSearch(block);
+
+  searchResults.style.display = 'block';
+  searchResults.innerHTML = '';
+
+  const results = await getLocaleBlogContents();
+  const prefixLength = getLocaleInfo().placeholdersPrefix.length;
+  const searchTerms = searchValue.toLowerCase().split(/\s+/);
+  const includedPaths = [];
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const result of results) {
+    const metaContents = `${result.title.toLowerCase()} ${result.description.toLowerCase()} ${result.header.toLowerCase()} ${result.path.substring(prefixLength).toLowerCase()}`;
+    if (searchTerms.some((term) => metaContents.includes(term))) {
+      const link = a({ href: result.path }, result.header);
+      markSearchTerms(link, searchTerms);
+      searchResults.append(link);
+      includedPaths.push(result.path);
+    }
+  }
+
+  // eslint-disable-next-line no-restricted-syntax
+  for (const result of results) {
+    if (!includedPaths.includes(result.path)) {
+      if (searchTerms.some((term) => result.content.toLowerCase()
+        .includes(term))) {
+        searchResults.append(
+          a({ href: result.path }, result.header),
+        );
+      }
+    }
+  }
+
+  unindicateSearch(block);
+  // while sorting by publicationDate would potentially make sense, it's not what's currently
+  // implemented on servicenow.com
```

Comment:
The contents of the `blog-content` sheet are already sorted descending based on the publication date, so when iterating through the list, you are already doing it starting from the latest blogs.

Reply:
removed the comment in 652ee2e